### ISSUE TITLE
Given exponent overflowing the maximum accepted scale (255).: TryFromIntError(())

### ISSUE THREAD

**Issue by hitzam-wartek** [prisma/prisma#15079]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.16.0           | 
| OS              | linux-musl         |
| Prisma Client   | 4.2.1              |
| Query Engine    | 2920a97877e12e055c1333079b8d19cee7f33826|
| Database        | sqlserver          |

## Logs
```
prisma:tryLoadEnv Environment variables not found at null
prisma:tryLoadEnv Environment variables not found at undefined
prisma:tryLoadEnv No Environment variables loaded
prisma:tryLoadEnv Environment variables not found at null
prisma:tryLoadEnv Environment variables not found at undefined
prisma:tryLoadEnv No Environment variables loaded
prisma:client dirname /opt/bku-sync-final/node_modules/@@internal/prisma/client/sqlserver/cabang
prisma:client relativePath ../../../../../../src/util/db/sqlserver
prisma:client cwd /opt/bku-sync-final/src/util/db/sqlserver
prisma:client clientVersion 4.2.1
prisma:client clientEngineType library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in /opt/bku-sync-final/node_modules/@@internal/prisma/client/.prisma/client
prisma:client:libraryEngine:loader Searching for Query Engine Library in /opt/bku-sync-final/node_modules/@@internal/prisma/client/sqlserver/cabang
prisma:client:libraryEngine:loader loadEngine using /opt/bku-sync-final/node_modules/@@internal/prisma/client/sqlserver/cabang/libquery_engine-linux-musl.so.node
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
```

## Client Snippet
```ts
// PLEASE FILL YOUR CODE SNIPPET HERE
dbContext.aktivasi_bku
                .upsert({
                    where: {
                        id_anggaran_id_periode: {
                            id_anggaran: data.id_anggaran,
                            id_periode: data.id_periode,
                        },
                    },
                    create: data,
                    update: data,
                })
```

## Schema
```prisma
// PLEASE ADD YOUR SCHEMA HERE IF POSSIBLE
model aktivasi_bku {
  id_anggaran            String       @db.Uuid
  id_periode             Decimal      @db.Decimal(2, 0)
  tanggal_aktivasi       DateTime     @db.Timestamp(6)
  tanggal_finish         DateTime?    @db.Timestamp(6)
  saldo_awal_bank        Decimal?     @db.Money
  saldo_awal_tunai       Decimal?     @db.Money
  saldo_akhir_bank       Decimal?     @db.Money
  saldo_akhir_tunai      Decimal?     @db.Money
  saldo_awal_bank_sisa   Decimal?     @db.Money
  saldo_awal_tunai_sisa  Decimal?     @db.Money
  saldo_akhir_bank_sisa  Decimal?     @db.Money
  saldo_akhir_tunai_sisa Decimal?     @db.Money
  soft_delete            Decimal      @db.Decimal(1, 0)
  create_date            DateTime     @db.Timestamp(6)
  last_update            DateTime     @db.Timestamp(6)
  updater_id             String?      @db.Uuid
  anggaran               anggaran     @relation(fields: [id_anggaran], references: [id_anggaran], onUpdate: Restrict, map: "FK_aktivasi_bku_anggaran_id_anggaran")
  ref_periode            ref_periode  @relation(fields: [id_periode], references: [id_periode], onUpdate: Restrict, map: "FK_aktivasi_bku_ref_periode_id_periode")
  report_bku             report_bku[]

  @@id([id_anggaran, id_periode], map: "PK_aktivasi_bku")
  @@index([id_periode], map: "IX_aktivasi_bku_id_periode")
}
```

## Prisma Engine Query
```
{"X":{}}
```

**Comment by hitzam-wartek** [comment]
the data tried to upsert:
```{
    "id_anggaran": "5b329d4e-3eab-4d21-95fa-820f23bfe1c2",
    "id_periode": 90,
    "tanggal_aktivasi": "2022-10-01T07:00:00.000Z",
    "tanggal_finish": "2022-10-31T07:00:00.000Z",
    "saldo_awal_bank": 14495000,
    "saldo_awal_tunai": 0,
    "saldo_akhir_bank": 13895000,
    "saldo_akhir_tunai": 0,
    "saldo_awal_bank_sisa": 0,
    "saldo_awal_tunai_sisa": 0,
    "saldo_akhir_bank_sisa": 0,
    "saldo_akhir_tunai_sisa": 0,
    "soft_delete": 0,
    "create_date": "2022-08-31T18:46:52.000Z",
    "last_update": "2022-08-31T19:43:22.000Z",
    "updater_id": "9776290b-c309-41c5-a339-568ad0425e9b"
}

@danstarns @Jolg42

**Comment by Jolg42** [comment]
Potentially related (same error message)
https://github.com/prisma/prisma/issues/14703

**Comment by pimeys** [comment]
It is our `BigDecimal` Rust library and a conversion we do in a way that generates a scale that does not fit to how SQL Server handles decimals. For some reasons a `Decimal(90, 0)` becomes `Decimal(9, -1)` sometimes with the crate. Not in all queries. The scale must be an unsigned 8-bit integer, so `-1` will not fit there. There's a fix for this coming in 4.4.0.

**Comment by hitzam-wartek** [comment]
nice info. thanks... 😃 👍
